specutils/io/default_loaders/tabular_fits.py

```diff
@@ -181,8 +187,12 @@ def tabular_fits_writer(spectrum, file_name, hdu=1, update_header=False, **kwarg
         if columns[c].ndim > 1:
             columns[c] = columns[c].T
 
-    tab = Table(columns, names=colnames, meta=header)
+    tab = Table(columns, names=colnames)
+    hdu1 = fits.BinTableHDU(data=tab, header=header)
```

Comment:
```suggestion
    hdu1 = fits.BinTableHDU(data=tab)
```
Now that the reader is by default always just reading from the primary header, perhaps just give this a try and see if it still passes the round-trip tests? Although to be fully compatible, the writer should probably then get a `store_data_header` option as well.